### ISSUE TITLE
Error when doing migration on new deploy (Schema dump style not set to sql)

### ISSUE THREAD

**Issue by zph** [diogob/activerecord-postgres-hstore#117]
Thanks for creating this adapter :D!! I'm enjoying getting to use hashes in Postgres!

Here's a small error report related to running migrations with hstore.

First off, seems like I should have switched to sql style schema dumps!

Second, hit this error below in the adapter and was able to dodge the issue by changing:

`/home/vagrant/.rvm/gems/ruby-2.0.0-p353/bundler/gems/activerecord-postgres-hstore-396f5af9af5e/lib/activerecord-postgres-hstore/activerecord.rb:158`

From

```
    if value && column && column.sql_type == 'hstore'
```

To

```
    if value && column && column.try(:sql_type) == 'hstore'
```

Version of gem is the current master from github (ie Gemfile is set to load activerecord-postgres-hstore from github)

tl;dr && Solution

This hack worked to get around this, but probably needs a more robust response by someone more familiar with the project :+1: .

Full stacktrace below:

```
vagrant@rails-dev-box:/vagrant/clerk/db/migrate$ be rake db:migrate
(in /vagrant/clerk)
[RailsAdmin] RailsAdmin initialization disabled by default. Pass SKIP_RAILS_ADMIN_INITIALIZER=false if you need it.
==  DeviseCreateUsers: migrating ==============================================
-- create_table(:users)
rake aborted!
An error has occurred, this and all later migrations canceled:

undefined method `sql_type' for #<ActiveRecord::ConnectionAdapters::PostgreSQLAdapter::ColumnDefinition:0xe459c38>/home/vagrant/.rvm/gems/ruby-2.0.0-p353/bundler/gems/activerecord-postgres-hstore-396f5af9af5e/lib/activerecord-postgres-hstore/activerecord.rb:158:in `quote_with_hstore'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract/schema_statements.rb:698:in `add_column_options!'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:169:in `add_column_options!'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/postgresql/schema_statements.rb:31:in `add_column_options!'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:136:in `visit_ColumnDefinition'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/postgresql/schema_statements.rb:14:in `visit_ColumnDefinition'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:117:in `accept'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:143:in `block in visit_TableDefinition'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:143:in `map'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:143:in `visit_TableDefinition'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:117:in `accept'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract/schema_statements.rb:190:in `create_table'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:625:in `block in method_missing'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:597:in `block in say_with_time'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:597:in `say_with_time'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:617:in `method_missing'
/vagrant/clerk/db/migrate/20131208080136_devise_create_users.rb:3:in `change'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:571:in `exec_migration'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:555:in `block (2 levels) in migrate'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:554:in `block in migrate'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:294:in `with_connection'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:553:in `migrate'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:709:in `migrate'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:959:in `block in execute_migration_in_transaction'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:1005:in `block in ddl_transaction'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:202:in `block in transaction'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:210:in `within_new_transaction'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:202:in `transaction'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/transactions.rb:209:in `transaction'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:1005:in `ddl_transaction'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:958:in `execute_migration_in_transaction'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:920:in `block in migrate'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:916:in `each'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:916:in `migrate'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:764:in `up'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/migration.rb:742:in `migrate'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/activerecord-4.0.2/lib/active_record/railties/databases.rake:42:in `block (2 levels) in <top (required)>'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/bin/ruby_executable_hooks:15:in `eval'
/home/vagrant/.rvm/gems/ruby-2.0.0-p353/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
vagrant@rails-dev-box:/vagrant/clerk/db/migrate$ vim ~/.rvm/gems/ruby-2.0.0-p353/bundler/gems/activerecord-postgres-hstore-396f5af9af5e/lib/activerecord-postgres-hstore/activerecord.rb +158
vagrant@rails-dev-box:/vagrant/clerk/db/migrate$ be rake db:migrate
(in /vagrant/clerk)
[RailsAdmin] RailsAdmin initialization disabled by default. Pass SKIP_RAILS_ADMIN_INITIALIZER=false if you need it.
==  DeviseCreateUsers: migrating ==============================================
-- create_table(:users)
   -> 1.4547s
   -- add_index(:users, :email, {:unique=>true})
   -> 0.0026s
   -- add_index(:users, :reset_password_token, {:unique=>true})
   -> 0.0021s
   ==  DeviseCreateUsers: migrated (1.4608s) =====================================
   '
```
